### ISSUE TITLE
Grafana 11: Performance issues due to high database load

### ISSUE THREAD

**Issue by arogozin** [grafana/grafana#89192]
### What happened?

We have recently upgraded to Grafana 11.0.0. It is used by a large number of users (> 1k) with a large number of dashboards / alerts (thousands). After upgrading to Grafana 11.0.0 we noticed a really large spike in database connection count. This caused brief period where users were unable to login or load dashboards. The error we saw at that time was: ```Failed to get namespaces visible to the user: [folder.internal] failed to fetch subfolders: Error 1040 (08004): Too many connections```
We adjusted max connection count, which resolved "too many connections" issue.

A few days later we started receiving reports for dashboards loading significantly slower and some endpoints used to retrieve alerts would take 2+ minutes instead of a couple of seconds in 10.4 version. While looking at slow queries, there were a few that stood out:

```
SELECT
  `dashboard` . `id`,
  `dashboard` . `uid`,
  `dashboard` . `title`,
  `dashboard` . `slug`,
  `dashboard_tag` . `term`,
  `dashboard` . `is_folder`,
  `dashboard` . `folder_id`,
  `folder` . `uid` AS `folder_uid`,
  `folder` . `title` AS `folder_slug`,
  `folder` . `title` AS `folder_title`
FROM (
  SELECT
    `dashboard` . `id`
  FROM
    `dashboard`
  WHERE
    ( ( ( `dashboard` . `org_id` = ?
          AND `dashboard` . `uid` IN (
          SELECT
            `d` . `uid`
          FROM
            `dashboard` `d`
          INNER JOIN
            `folder` `f1`
          ON
            `d` . `uid` = `f1` . `uid`
            AND `d` . `org_id` = `f1` . `org_id`
          WHERE
            `f1` . `org_id` = ?
            AND `f1` . `uid` IN (
            SELECT
              `identifier`
            FROM
              `permission`
            WHERE
              `kind` = ?
              AND `attribute` = ?
              AND `role_id` IN (
              SELECT
                `id`
              FROM
                `role`
              INNER JOIN (
                SELECT
                  `ur` . `role_id`
                FROM
                  `user_role` AS `ur`
                WHERE
                  `ur` . `user_id` = ?
                  AND ( `ur` . `org_id` = ?
                    OR `ur` . `org_id` = ? ) UNION
                SELECT
                  `br` . `role_id`
                FROM
                  `builtin_role` AS `br`
                WHERE
                  `br` . `role` IN (?)
                  AND ( `br` . `org_id` = ?
                    OR `br` . `org_id` = ? ) ) AS `all_role`
              ON
                `role` . `id` = `all_ro......
(truncated)
```

On average it took 5.27 hr to execute the query and it scanned over ten million rows and returned a couple hundred thousand results.

There are a few more queries that took significantly longer to execute as well:

```SELECT
  `dashboard` . `id`,
  `dashboard` . `uid`,
  `dashboard` . `title`,
  `dashboard` . `slug`,
  `dashboard_tag` . `term`,
  `dashboard` . `is_folder`,
  `dashboard` . `folder_id`,
  `folder` . `uid` AS `folder_uid`,
  `folder` . `title` AS `folder_slug`,
  `folder` . `title` AS `folder_title`
FROM (
  SELECT
    `dashboard` . `id`
  FROM
    `dashboard`
  WHERE
    ( ( `dashboard` . `uid` IN (
        SELECT
          `identifier`
        FROM
          `permission`
        WHERE
          `kind` = ?
          AND `attribute` = ?
          AND `role_id` IN (
          SELECT
            `id`
          FROM
            `role`
          INNER JOIN (
            SELECT
              `ur` . `role_id`
            FROM
              `user_role` AS `ur`
            WHERE
              `ur` . `user_id` = ?
              AND ( `ur` . `org_id` = ?
                OR `ur` . `org_id` = ? ) UNION
            SELECT
              `br` . `role_id`
            FROM
              `builtin_role` AS `br`
            WHERE
              `br` . `role` IN (?)
              AND ( `br` . `org_id` = ?
                OR `br` . `org_id` = ? ) ) AS `all_role`
          ON
            `role` . `id` = `all_role` . `role_id` )
          AND ACTION = ? )
        AND NOT `dashboard` . `is_folder` )
      OR ( ( `dashboard` . `org_id` = ?
          AND `dashboard` . `folder_id` IN (
          SELECT
            `d` . `id`
          FROM
            `dashboard` `d`
          INNER JOIN
            `folder` `f1`
          ON
            `d` ......
(truncated)
```

On average it took over 6 minutes to execute the query and it scanned over million rows and returned a couple thousand results.

After downgrade back to Grafana 10.4, database restore, then individual version upgrade up until Grafana 11.0.0 we still see very high CPU usage on MySQL db instances (>90%). We are interested to see if there are any additional reports or other insights that can be provided here.

### What did you expect to happen?

Upgrade to Grafana 11.0.0 to have no performance impact to our users.

### Did this work before?

Yes, there were no performance issues related to high CPU usage in database or SIGNIFICANTLY SLOW queries described above.

### How do we reproduce it?

1. Run Grafana 10.4 with thousands of active users and thousands of dashboards and alerts.
2. Upgrade to Grafana 11.0.0
3. Load dashboards, search for dashboards as non-admin user (we are aware of that as separate issue as well, but it was significantly degraded after upgrade), try requesting alerts via api.

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: 11.0.0
OS: Docker image running grafana/grafana-oss:11.0.0
DB: MySQL (CloudSQL large installation of instances in GCP)
Browser: Chrome

### Grafana platform?

Docker

**Comment by marefr** [comment]
@grafana/grafana-search-and-storage please triage

**Comment by sjoeboo** [comment]
Dashboards: about 10k,though about 4k come and go daily (Pull Request review generated) 
Roles: every user is Editor. a handful of admins, but thats it.
Folders: 2500 roughly, no subfolders (we haven't specifically enabled/disabled them, but our dshboards/alerts as code tool doesn't support them yet so there should be none)

**Comment by timlevett** [comment]
Just to try a workaround for now, if you disable the feature flag `nestedFolders` ([ref](https://github.com/grafana/grafana/blob/main/pkg/services/featuremgmt/registry.go#L254)) which is on by default in 11.0.0, does that remove the performance issue for you? [Here](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/) are the docs on configuring feature toggles.

**Comment by sjoeboo** [comment]
We config via env vars, would that be 

`ENV GF_FEATURE_TOGGLES_ENABLE_NESTED_FOLDERS=false` 

or

`ENV GF_FEATURE_TOGGLES_ENABLENESTEDFOLDERS=false` ? 

(side note, since feature toggles are enabled like 

`ENV GF_FEATURE_TOGGLES_ENABLE='something somethingelse'` 
it would be nice to be consistent ie
`ENV GF_FEATURE_TOGGLES_DISABLE='nestedFolders'`

**Comment by DanCech** [comment]
`ENV GF_FEATURE_TOGGLES_NESTEDFOLDERS=false` should do the trick, or you can specify it in your custom.ini:
```
[feature_toggles]
nestedFolders = false
```

**Comment by papagian** [comment]
@arogozin thank for reporting this
could you share the full queries: these are used for different purposes and they slightly differ from case to case? do you happen to know if those are consistently slow or occasionally?
also,  do you know how many dashboard tags do you have (this table also used by the query)
it could be helpful to debug if you could provide also the grafana configuration and any relevant logs

**Comment by sjoeboo** [comment]
We'll work on getting the exact queries being run, but i expect its a wide spread of dashboard searches both from users in the UI and via API when our dashboards-as-code tooling is looking for dashboards to update/replace. 

Tags: about 8300 currently. 

Config (via env vars) 
```
FROM grafana/grafana-oss:11.0.0
LABEL maintainer "abc"

COPY grafana.ini /etc/grafana/grafana.ini

ENV GF_SERVER_DOMAIN=monitoring.abc.net
#ENV GF_SERVER_ROOT_URL=%(protocol)s://%(domain)s/
ENV GF_SERVER_ROOT_URL=https://monitoring.abc.net
ENV GF_SERVER_ENABLE_GZIP=true
ENV GF_ROUTER_LOGGING=false

ENV GF_SNAPSHOTS_EXTERNAL_ENABLED=false
ENV GF_USERS_AUTO_ASSIGN_ORG=true
ENV GF_USERS_AUTO_ASSIGN_ORG_ID=1
ENV GF_USERS_AUTO_ASSIGN_ORG_ROLE=Editor
ENV GF_AUTH_DISABLE_LOGIN_FORM=true
#ENV GF_AUTH_DISABLE_SIGNOUT_MENU=true

#ENV GF_AUTH_PROXY_ENABLED=true
#ENV GF_AUTH_PROXY_HEADER_NAME=SSO-User
#ENV GF_AUTH_PROXY_HEADERS="Email:SSO-Mail"


ENV GF_AUTH_GOOGLE_ENABLED=true
ENV GF_AUTH_GOOGLE_SCOPES='https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email'
ENV GF_AUTH_GOOGLE_AUTH_URL='https://accounts.google.com/o/oauth2/auth'
ENV GF_AUTH_GOOGLE_TOKEN_URL='https://accounts.google.com/o/oauth2/token'
ENV GF_AUTH_GOOGLE_ALLOWED_DOMAINS='abc.com'
ENV GF_AUTH_GOOGLE_ALLOW_SIGN_UP=true
#always just assign editor from GF_USERS_AUTO_ASSIGN_ORG_ROLE
ENV GF_AUTH_GOOGLE_SKIP_ORG_ROLE_SYNC=true
ENV GF_AUTH_GOOGLE_OAUTH_SKIP_ORG_ROLE_UPDATE_SYNC=true

ENV GF_AUTH_GOOGLE_CLIENT_ID=abc.apps.googleusercontent.com




ENV GF_DATABASE_URL=mysql://grafana:grafana@127.0.0.1:3306/grafana
ENV GF_DATABASE_TYPE=mysql
ENV GF_DATABASE_MAX_OPEN_CONN=100
ENV GF_DATABASE_MAX_IDLE_CONN=25
ENV GF_DATABASE_CONN_MAX_LIFETIME=3600
ENV GF_DATABASE_LOG_QUERIES=false

ENV GF_REMOTE_CACHE_TYPE=memcached
ENV GF_REMOTE_CACHE_CONNSTR=memcached-vanilla:11211

ENV GF_PATHS_PROVISIONING=/etc/grafana/provisioning

#ENV GF_ANALYTICS_GOOGLE_ANALYTICS_UA_ID=UA-48549177-4

ENV GF_ANALYTICS_GOOGLE_ANALYTICS_4_ID=G-YM1F727V9Q

ENV GF_DATAPROXY_LOGGING=false
ENV GF_DATAPROXY_TIMEOUT=600
ENV GF_DATAPROXY_SEND_USER_HEADER=true

ENV GF_METRICS_ENABLED=true

ENV GF_DASHBOARDS_MIN_REFRESH_INTERVAL=30s

#ENV GF_UNIFIED_ALERTING_ENABLED=false
#ENV GF_ALERTING_ENABLED=true

#Disable alert evaluation (alerting instances override)
ENV GF_UNIFIED_ALERTING_EXECUTE_ALERTS=false
# Set min alert evaluation interval
ENV GF_UNIFIED_ALERTING_MIN_INTERVAL=5m
ENV GF_UNIFIED_ALERTING_EVALUATION_TIMEOUT=5m
ENV GF_UNIFIED_ALERTING_MAX_ATTEMPTS=5

ENV GF_ALERTING_NODATA_OR_NULLVALUES="ok"
ENV GF_ALERTING_ERROR_OR_TIMEOUT="ok"

# Move to alerting-only instance
#ENV GF_ALERTING_MAX_ANNOTATION_AGE=1d
#ENV GF_ALERTING_MAX_ANNOTATIONS_TO_KEEP=100000

# Clean up annotations in addition to the external cron
ENV GF_ANNOTATIONS_API_MAX_AGE=7d
ENV GF_ANNOTATIONS_API_MAX_ANNOTATIONS_TO_KEEP=50000
ENV GF_ANNOTATIONS_DASHBOARD_MAX_AGE=7d
ENV GF_ANNOTATIONS_DASHBOARD_MAX_ANNOTATIONS_TO_KEEP=50000

ENV GF_SMTP_ENABLED=true
ENV GF_SMTP_HOST=smtp.abc.net:2525
ENV GF_SMTP_USER=apikey
#ENV GF_SMTP_PASSWORD
ENV GF_SMTP_FROM_ADDRESS=alert@abc.net


RUN grafana-cli plugins install grafana-bigquery-datasource 1.2.9
RUN grafana-cli plugins install googlecloud-logging-datasource 1.2.1
RUN grafana-cli plugins install grafana-sentry-datasource 1.2.1
RUN grafana-cli plugins install servicenow-cloudobservability-datasource 3.3.0
RUN grafana-cli plugins install marcusolsson-json-datasource 1.3.10

ENV GF_LOG_LEVEL=debug

ENV GF_LOG_CONSOLE_FORMAT=json

ENV GF_SECURITY_DISABLE_INITIAL_ADMIN_CREATION=true
ENV GF_SECURITY_ALLOW_EMBEDDING=true


#Feature Toggles
ENV GF_FEATURE_TOGGLES_ENABLE='prometheusResourceBrowserCache prometheusMetricEncyclopedia correlations featureToggleAdminPage'
ENV GF_FEATURE_TOGGLES_NESTED_FOLDERS=false
ENV GF_FEATURE_TOGGLES_NESTEDFOLDERS=false
```

I'll note, we've also long been suffering from the "slow dashboard search" issue that I know will be fixed in 11.1, where non-admin users see VERY poor dashboard search performance. I'm wondering if this would be related (since the queries seem dashboard search based, yet us admins, while seeing our DB CPU usage sky high, do not suffer any issues in Grafana itself. 

Furthermore: disabling subfolders has no impact. Things were quiet over the weekend but as soon as our Grafana load of users/api acess returned Monday morning, CPU usage on the DB skyrocketed (i'll note this is a 96core CloudSQL instance qhich for years under the same load has been mostly idle CPU wise).

**Comment by bergquist** [comment]
I'd strongly recommend setting these value to the value number. 

```
ENV GF_DATABASE_MAX_OPEN_CONN=100
ENV GF_DATABASE_MAX_IDLE_CONN=25
```

With this configuration the grafana process will recycle a lot of connections during small spikes. I goes to 100. then down to needing 30 and at that point it can only keep 25 of them idle which means the rest are recycled. 

Not saying this is the root cause for the problem. but something to tweak :)

**Comment by sjoeboo** [comment]
As in, set both to 100 (idle==max) ?

**Comment by bergquist** [comment]
> As in, set both to 100 (idle==max) ?

Yes! Depending on how many replicas you have 100 might also be very low for that database size.

**Comment by sjoeboo** [comment]
We have 18 replicas in total (3x5 user facing, 2 alerting, and 1 view-only)

**Comment by sjoeboo** [comment]
example query log from slow query log:

```# Time: 2024-06-17T13:03:44.265710Z
# User@Host: grafana[grafana] @  [cloudsqlproxy~35.240.91.152]  thread_id: 65567  server_id: 4074031869
# Query_time: 6033.567913  Lock_time: 0.000667 Rows_sent: 489  Rows_examined: 31878614
SET timestamp=1718629424;
SELECT
			dashboard.id,
			dashboard.uid,
			dashboard.title,
			dashboard.slug,
			dashboard_tag.term,
			dashboard.is_folder,
			dashboard.folder_id,
			folder.uid AS folder_uid,
		
			folder.title AS folder_slug,
			folder.title AS folder_title  FROM ( SELECT dashboard.id FROM dashboard WHERE ((dashboard.uid IN (SELECT identifier FROM permission WHERE kind = 'dashboards' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read') AND NOT dashboard.is_folder) OR ((dashboard.org_id = 1 AND dashboard.folder_id IN (SELECT d.id FROM dashboard d INNER JOIN folder f1 ON d.uid = f1.uid AND d.org_id = f1.org_id  WHERE f1.org_id = 1 AND f1.uid IN (SELECT identifier FROM permission WHERE kind = 'folders' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read'))) OR (dashboard.org_id = 1 AND dashboard.folder_id IN (SELECT d.id FROM dashboard d INNER JOIN folder f1 ON d.uid = f1.uid AND d.org_id = f1.org_id  INNER JOIN folder f2 ON f1.parent_uid = f2.uid AND f1.org_id = f2.org_id  WHERE f2.org_id = 1 AND f2.uid IN (SELECT identifier FROM permission WHERE kind = 'folders' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read'))) OR (dashboard.org_id = 1 AND dashboard.folder_id IN (SELECT d.id FROM dashboard d INNER JOIN folder f1 ON d.uid = f1.uid AND d.org_id = f1.org_id  INNER JOIN folder f2 ON f1.parent_uid = f2.uid AND f1.org_id = f2.org_id  INNER JOIN folder f3 ON f2.parent_uid = f3.uid AND f2.org_id = f3.org_id  WHERE f3.org_id = 1 AND f3.uid IN (SELECT identifier FROM permission WHERE kind = 'folders' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read'))) OR (dashboard.org_id = 1 AND dashboard.folder_id IN (SELECT d.id FROM dashboard d INNER JOIN folder f1 ON d.uid = f1.uid AND d.org_id = f1.org_id  INNER JOIN folder f2 ON f1.parent_uid = f2.uid AND f1.org_id = f2.org_id  INNER JOIN folder f3 ON f2.parent_uid = f3.uid AND f2.org_id = f3.org_id  INNER JOIN folder f4 ON f3.parent_uid = f4.uid AND f3.org_id = f4.org_id  WHERE f4.org_id = 1 AND f4.uid IN (SELECT identifier FROM permission WHERE kind = 'folders' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read'))) OR (dashboard.org_id = 1 AND dashboard.folder_id IN (SELECT d.id FROM dashboard d INNER JOIN folder f1 ON d.uid = f1.uid AND d.org_id = f1.org_id  INNER JOIN folder f2 ON f1.parent_uid = f2.uid AND f1.org_id = f2.org_id  INNER JOIN folder f3 ON f2.parent_uid = f3.uid AND f2.org_id = f3.org_id  INNER JOIN folder f4 ON f3.parent_uid = f4.uid AND f3.org_id = f4.org_id  INNER JOIN folder f5 ON f4.parent_uid = f5.uid AND f4.org_id = f5.org_id  WHERE f5.org_id = 1 AND f5.uid IN (SELECT identifier FROM permission WHERE kind = 'folders' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read'))) OR (dashboard.org_id = 1 AND dashboard.folder_id IN (SELECT d.id FROM dashboard d INNER JOIN folder f1 ON d.uid = f1.uid AND d.org_id = f1.org_id  INNER JOIN folder f2 ON f1.parent_uid = f2.uid AND f1.org_id = f2.org_id  INNER JOIN folder f3 ON f2.parent_uid = f3.uid AND f2.org_id = f3.org_id  INNER JOIN folder f4 ON f3.parent_uid = f4.uid AND f3.org_id = f4.org_id  INNER JOIN folder f5 ON f4.parent_uid = f5.uid AND f4.org_id = f5.org_id  INNER JOIN folder f6 ON f5.parent_uid = f6.uid AND f5.org_id = f6.org_id  WHERE f6.org_id = 1 AND f6.uid IN (SELECT identifier FROM permission WHERE kind = 'folders' AND attribute = 'uid' AND role_id IN(SELECT id FROM role INNER JOIN (
			SELECT ur.role_id
			FROM user_role AS ur
			WHERE ur.user_id = 368
			AND (ur.org_id = 1 OR ur.org_id = 0)
		UNION
			SELECT br.role_id FROM builtin_role AS br
			WHERE br.role IN ('Editor')
			AND (br.org_id = 1 OR br.org_id = 0)
		) as all_role ON role.id = all_role.role_id)  AND action = 'annotations:read'))) AND NOT dashboard.is_folder)) AND dashboard.org_id=1 ORDER BY dashboard.title ASC LIMIT 1000 OFFSET 0) AS ids
		INNER JOIN dashboard ON ids.id = dashboard.id
LEFT OUTER JOIN folder ON folder.uid = dashboard.folder_uid AND folder.org_id = dashboard.org_id
	LEFT OUTER JOIN dashboard_tag ON dashboard.id = dashboard_tag.dashboard_id
 ORDER BY dashboard.title ASC;
```

also see lots of queries like this in the slow query log

```
# Time: 2024-06-17T13:04:13.038113Z
# User@Host: grafana[grafana] @  [cloudsqlproxy~35.240.91.152]  thread_id: 67655  server_id: 4074031869
# Query_time: 18.869277  Lock_time: 0.000064 Rows_sent: 274685  Rows_examined: 549370
SET timestamp=1718629453;
SELECT `record_key`, `provenance` FROM `provenance_type` WHERE (record_type = 'alertRule' AND org_id = 1) ORDER BY `id` DESC;
```

**Comment by papagian** [comment]
which MySQL version do you use? from the above query I assume that it's one not supporting recursive queries
can you confirm?

**Comment by sjoeboo** [comment]
Its MySQL 5.7.44.

**Comment by sjoeboo** [comment]
On a hunch I set `max_execution_time` in our DB to 2m (was default 0 == unlimited) as any heavy query seemed to be much longer than that, with the vast majority of non-problematic queries being well under that. My assumption was/is:

Slow query happens, runs for a long time. 
During this time, more of these come in. 
Each one opens lots of tables and keeps them open. 
More and more queries "build up" and are running concurrently. 
Eventually this starts to increase CPU load, open tables, and memory, until the DB is pretty bogged down with queries (which IMO seem like dashboard searches) which will never return to users anyways. 

This has seemed to have a positive impact on CPU load and thus open tables and memory consumption. 

But obviously a stop-gap as the queries shouldn't be running so long in the first place. 

Could this be related to https://github.com/grafana/grafana/pull/88019 and its those slow queries piling up?

**Comment by papagian** [comment]
> Its MySQL 5.7.44.

this version does not support recursive queries therefore the search query fallbacks using table joins instead for nested folder support
I'm a afraid that it's not going to scale with that big number of dashboards/folders
in fact, we [plan](https://github.com/grafana/grafana/issues/68446) to deprecate it since it has reached EOL
do you think you could upgrade?

**Comment by sjoeboo** [comment]
Upgrading to MySQL has had a really positive impact. 

I'd love it if we can have this added to the documentation as I don't see anything in either the 11 configuration docs about MySQL version recommendations nor anything in the "what's new"/"breaking changes" docs

Regardless, thanks for the help tracking this one down, it also helped ( a bit) with slow dashboard searches for our users, though we're still awaiting the "real fix" in 11.1.

**Comment by scottlepp** [comment]
Happy to hear upgrading MySQL helped.

The Grafana setup docs [supported databases](https://grafana.com/docs/grafana/latest/setup-grafana/installation/#supported-databases) shows MySQL 8+ is supported.  

Grafana 11.1 addressed show dashboard searches.  Closing this due to inactivity.  Please reopen if 11.1 didn't address the problem.
